lib/node_modules/@stdlib/_tools/eslint/rules/jsdoc-example-require-spacing/lib/main.js

```diff
@@ -0,0 +1,172 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var parseJSDoc = require( 'doctrine' ).parse;
+var trim = require( '@stdlib/string/base/trim' );
+var isObject = require( '@stdlib/assert/is-object' );
+var isEmptyArray = require( '@stdlib/assert/is-empty-array' );
+var findJSDoc = require( '@stdlib/_tools/eslint/utils/find-jsdoc' );
+
+
+// VARIABLES //
+
+var RE_NEWLINE = /\r?\n/g;
+var DOPTS = {
+	'sloppy': true,
+	'unwrap': true,
+	'tags': [ 'example' ]
+};
+var rule;
+
+
+// MAIN //
+
+/**
+* Rule to enforce empty lines between requires and code in JSDoc examples.
+*
+* @param {Object} context - ESLint context
+* @returns {Object} rule object
+*/
+function main( context ) {
+	var source = context.getSourceCode();
+
+	/**
+	* Reports the error message.
+	*
+	* @private
+	* @param {Object} node - node to report
+	*/
+	function report( node ) {
+		context.report({
+			'node': node,
+			'message': 'Missing empty line between require statement and code in JSDoc example'
+		});
+	}
+
+	/**
+	* Checks if a line contains a require statement.
+	*
+	* @private
+	* @param {string} line - line to check
+	* @returns {boolean} boolean indicating if line contains a require statement
+	*/
+	function hasRequire( line ) {
+		return trim( line ).indexOf( 'require(' ) !== -1;
+	}
+
+	/**
+	* Checks if a line is empty.
+	*
+	* @private
+	* @param {string} line - line to check
+	* @returns {boolean} boolean indicating if line is empty
+	*/
+	function isEmpty( line ) {
```

Comment:
@Planeshifter This function can be moved to parent scope.

Reply:
Done.